Resfeber/CoreData/Model/Trip+CoreDataProperties.swift

```diff
@@ -43,5 +43,21 @@ extension Trip {
 }
 
 extension Trip : Identifiable {
-
+    var eventsArray: [Event] {
+        let eventsArray = events?.allObjects as? [Event] ?? []
+        
+        /// Events sorted by date; events without a date are positioned at the end
+        return eventsArray.sorted { (x, y) -> Bool in
+            
+            if let xStartDate = x.startDate {
+                if let yStartDate = y.startDate {
+                    return xStartDate < yStartDate
+                } else {
+                    return true
+                }
+            }
+            
+            return y.startDate == nil
+        }
+    }
```

Comment:
It would probably be good to update the logic here to fallback to the endDate if the events you are looking at don't have a start date. That way you sort by start date, then end date, then no date.

Reply:
If you want some code that you can put in a playground to test out the logic, you can use this
```
struct Event {
    var name: String
    var startDate: Int?
    var endDate: Int?
}

let events = [
    Event(name: "event a", startDate: 1, endDate: 3),
    Event(name: "event b", startDate: 1, endDate: nil),
    Event(name: "event c", startDate: 3, endDate: 5),
    Event(name: "event d", startDate: nil, endDate: 10),
    Event(name: "event e", startDate: 2, endDate: 4),
    Event(name: "event f", startDate: nil, endDate: nil)
]
var eventsArray: [Event] {

    /// Events sorted by date; events without a date are positioned at the end
    return events.sorted { (x, y) -> Bool in
        
        if let xStartDate = x.startDate {
            if let yStartDate = y.startDate {
                return xStartDate < yStartDate
            } else {
                return true
            }
        }
        
        return y.startDate == nil
    }
}

print(eventsArray.map( { $0.name }))
```

Reply:
Good idea. I refactored the sorting code to sort by end date in case an event does not have a start date specified.